source/material_model/melt_phipps_morgan.cc

```diff
@@ -0,0 +1,1363 @@
+                  // when above the liquidus Q!
+                  double enthalpy_of_fusion = Fe_enthalpy_of_fusion * bulk_composition + Mg_enthalpy_of_fusion * (1.0-bulk_composition);
+                  enthalpy_of_fusion /= melt_molar_mass;
+
+                  enthalpy_out->enthalpies_of_fusion[q] = enthalpy_of_fusion;
+                }
+
+              // cutoff for viscosity at 30%
+              const double porosity = std::min(0.3, std::max(in.composition[q][porosity_idx],0.0));
+              //out.viscosities[q] = (1.0 - porosity) * eta_0 * exp(- alpha_phi * porosity);
+              out.viscosities[q] = (1.0 - porosity) * exp(- alpha_phi * porosity);
+            }
+          else
+            {
+              //out.viscosities[q] = eta_0;
+              out.viscosities[q] = 1.;
+
+              // no melting/freezing is used in the model --> set all reactions to zero
+              for (unsigned int c=0; c<in.composition[q].size(); ++c)
+                {
+                  if (reaction_rate_out != nullptr)
+                    reaction_rate_out->reaction_rates[q][c] = 0.0;
+                }
+            }
+
+          out.entropy_derivative_pressure[q]    = 0.0;
+          out.entropy_derivative_temperature[q] = 0.0;
+          out.thermal_conductivities[q] = thermal_conductivity;
+
+          for (unsigned int c=0; c<in.composition[q].size(); ++c)
+            out.reaction_terms[q][c] = 0.0;
+
+          const double delta_temp = in.temperature[q]-this->get_adiabatic_conditions().temperature(in.position[q]);
+          const double visc_temperature_dependence = std::max(std::min(std::exp(-thermal_viscosity_exponent*delta_temp/this->get_adiabatic_conditions().temperature(in.position[q])),1e8),1e-8);
+          out.viscosities[q] *= visc_temperature_dependence;
+        }
+
+      // fill melt outputs if they exist
+
+      if (melt_out != nullptr)
+        {
+          const unsigned int porosity_idx = this->introspection().compositional_index_for_name("porosity");
+
+          for (unsigned int q=0; q<in.position.size(); ++q)
+            {
+              double porosity = std::max(in.composition[q][porosity_idx],0.0);
+
+              melt_out->fluid_viscosities[q] = eta_f;
+              melt_out->permeabilities[q] = reference_permeability * std::pow(porosity,3) * std::pow(1.0-porosity,2);
+
+              // limit porosity to disaggregation threshold
+              porosity = std::min(0.3, porosity);
+
+              const double porosity_threshold = 0.01 * std::pow(this->get_melt_handler().melt_parameters.melt_scaling_factor_threshold, 1./3.);
+              melt_out->compaction_viscosities[q] = (1.0 - porosity) * xi_0 / std::max(porosity, porosity_threshold);
+
+              const double delta_temp = in.temperature[q]-this->get_adiabatic_conditions().temperature(in.position[q]);
+              // Q! Why this particular cutoff 1e-4-1e4?
```

Comment:
You would need to ask Juliane about this, but I suspect it's because melting sometimes led to a very large temperature dependence on the viscosity, and that was a problem for convergence.